src/app/components/operations/login/login.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { LoginUser } from 'src/app/models/login-user';
+import { AuthService } from 'src/app/services/auth.service';
+import { TokenService } from 'src/app/services/token.service';
+
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  isLogged = false;
+  isLoginFail = false;
+  loginUser: LoginUser;
+  roles : string[] = [];
+  messageError: string;
+  formLogin: FormGroup;
+
+  constructor(
+    private tokenService: TokenService,
+    private authService: AuthService,
+    private router: Router,
+    private formBuilder: FormBuilder
+  ) { 
+    this.formLogin = formBuilder.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required]
+    })
+
+  }
+
+  ngOnInit(): void {
+    if(this.tokenService.getToken()){
+      this.isLogged = true;
+      this.isLoginFail = false;
+      this.roles = this.tokenService.getAuthorities();
+    }
+  }
+
+
+  onLogin(){
+    this.loginUser = new LoginUser(this.formLogin.value.username, this.formLogin.value.password);
+    console.log(this.loginUser)
+    this.authService.login(this.loginUser).subscribe(
+      response => {
+        var res = response
+        //var res = response[0]
+        console.log(response)
+        this.isLogged = true;
+        this.isLoginFail = false;
+
+        console.log("User token: " + res.token);
+        console.log("Username: " + res.username);
+        console.log("User roles: " + res.authorities);
+
+        this.tokenService.setToken(res.token);
+        this.tokenService.setUsername(res.username);
+        this.tokenService.setAuthorities(res.authorities);
+
+        console.log("User " + res.username + "  logged sucessfully.")
+
+        this.router.navigate(['/']);
+        
+      }, err =>{
```

Comment:
Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/ispp-yourney/angular-app/pullRequest?prid=7122565)
```suggestion
      }, err => {
```